monad-rpc/src/handlers/mod.rs

```diff
@@ -263,25 +269,34 @@ async fn debug_getRawTransaction(
 
 #[allow(non_snake_case)]
 async fn debug_traceBlockByHash(
-    _: RequestId,
+    request_id: RequestId,
     app_state: &MonadRpcResources,
     params: RequestParams<'_>,
 ) -> Result<Box<RawValue>, JsonRpcError> {
     let triedb_env = app_state.triedb_reader.as_ref().method_not_supported()?;
-    let params = serde_json::from_str(params.get()).invalid_params()?;
+    let params: MonadDebugTraceBlockByHashParams =
+        serde_json::from_str(params.get()).invalid_params()?;
+    if params.requires_replay() {
+        return collect_debug_trace_via_replay(request_id, triedb_env, app_state, &params).await;
```

Comment:
The serialized result is returning incorrectly. We want to serialize the T, not Result<T,E>.  See below where we `.map(serialize_result)?`

Reply:
Thanks! I've applied `.map(serialize_result)?` to the three instances now.